src/main/java/io/github/tjheslin1/patterdale/metrics/ScrapeProbe.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2018 Thomas Heslin <tjheslin1@gmail.com>.
+ *
+ * This file is part of Patterdale-jvm.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.github.tjheslin1.patterdale.metrics;
+
+import io.github.tjheslin1.patterdale.metrics.probe.OracleSQLProbe;
+import io.github.tjheslin1.patterdale.metrics.probe.ProbeResult;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.Callable;
+
+public class ScrapeProbe implements Callable<List<ProbeResult>> {
```

Comment:
Instead of having `equals` and `hashCode` you can extend [ValueType](https://github.com/tjheslin1/Patterdale/blob/master/src/main/java/io/github/tjheslin1/patterdale/ValueType.java#L26) which gives you these methods for free. I've tested this on your fork and it works fine 👍

Reply:
I thought about this but decided against it as the ScrapeProbe class isn't really a ValueType. I looked into using the Apache reflection based utility methods used in ValueType in this class, though I thought it seemed a bit inefficient to use reflection when there is only one field.

In truth, these methods were only added to be able to mock the ExecutorService with the correct probe values in MetricsUseCaseTest, so they're not really called anyway! If you would prefer extending from ValueType then I will do that :+1:

Reply:
Good point 🤔 
I definitely see what you mean. I would say for conciseness that `extends ValueType` would be nice to have, even though it's not exactly the right fit. I can see from running the tests that an implementation is necessary.
So if you could make the change, much appreciated 👍

Reply:
Usually when this kind of thing happens (needing equality for testability because of e.g. mockito when matching) it is a sign that the tests are very coupled to the implementation and not the intention of them. Probably the tests could be written more cleanly with the unit under test being a cluster of real classes and testing the end result rather than testing the intermediate interactions between many mocks. Just the usual behaviour vs data driven testing trade off.

Reply:
@theangrydev I'll have a look at the testing separate to this PR 👍 I definitely agree. Identifying those boundaries is something I've been working on in my current team.